src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -305,10 +324,19 @@ impl AstDisplay for Connector {
                 f.write_node(&display::escape_single_quote_string(path));
                 f.write_str("'");
             }
-            Connector::S3 { bucket, pattern } => {
+            Connector::S3 {
+                bucket,
+                key_sources,
+                pattern,
+            } => {
                 f.write_str("S3 BUCKET '");
                 f.write_str(&display::escape_single_quote_string(bucket));
-                f.write_str("' OBJECTS FROM SCAN");
+                f.write_str("' OBJECTS FROM");
+                for ks in key_sources {
+                    match ks {
+                        KeySource::Scan => f.write_str(" SCAN"),
+                    }
```

Comment:
For parity with the other AST nodes, `KeySource` ought to implement `AstDisplay`, and then this method can delegate to that.